dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -0,0 +1,339 @@
+package definitions
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"errors"
+
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/loadbalancer"
+	"gopkg.in/yaml.v2"
+)
+
+type RouteGroupValidator struct {
+	FilterRegistry filters.Registry
+}
+
+var rgv = &RouteGroupValidator{}
+
+func (rgv *RouteGroupValidator) Validate(item *RouteGroupItem) error {
+	err := rgv.BasicValidation(item)
+	if rgv.FilterRegistry != nil {
+		err = errors.Join(err, rgv.FiltersValidation(item))
+	}
+	return err
+}
+
+func (rgv *RouteGroupValidator) BasicValidation(item *RouteGroupItem) error {
```

Comment:
Do we need this exported? Lets make it unexported, we can always export later when needed.

Reply:
Lets unexport also all other new methods

Reply:
Exported so if user want to run it individually.